libc/resolve_path.c

```diff
@@ -475,6 +475,48 @@ TEST(resolve_path, symlink_rename)
 }
 
 
+/* check the behavior of the realpath function when it encounters pathname resolution of a symbolic link that result with a huge length */
+TEST(resolve_path, symlink_long_resolution)
+{
+	const char symNameShort[] = "symShort";
+	char symNameLong[NAME_MAX / 2];
+	char resolved[PATH_MAX + 1];
+	char basePath[PATH_MAX - 5] = "symShort/";
+	int basePathLen = strlen(basePath);
+	const char *pathSegment = "dev/../";
```

Comment:
Not so important, but imho when we don't need to allocate more space, the pointer is ok. To increase readability we could place it next to the second declared string.

```suggestion
	char symNameLong[NAME_MAX / 2];
	char resolved[PATH_MAX + 1];
	char basePath[PATH_MAX - 5] = "symShort/";
	int basePathLen = strlen(basePath);
        const char *symNameShort = "symShort";
	const char *pathSegment = "dev/../";
```

Reply:
Not so crucial, so we can leave it, but I meant to place it a few lines lower.